Add Persistence strategy at Workflow / Activity Level

jdevillard opened the pull request:
This PR add the necessary UI to configure Persitence Strategy at Workflow and Activity Level

#118

jdevillard committed:
Add Persistence strategy at Workflow / Activity Level

jdevillard committed:
Merge remote-tracking branch 'origin/main' into features/activity-log-record-storage

jdevillard committed:
remove using about removed Contracts

jdevillard committed:
refactor and renaming to LogPersistenceMode

sfmskywalker committed:
Merge remote-tracking branch 'origin/main' into features/activity-log-record-storage

sfmskywalker committed:
Refactor PersistenceTab and update project references

Updated various aspects of the PersistenceTab class and its functionality to improve code quality and readability. Simplified the handling of persistence configurations and simplified the use of properties. Transitioned project reference for Elsa.Api.Client from package reference to direct project reference for better development experience in Elsa.Studio.Core.

jdevillard committed:
Merge remote-tracking branch 'origin/main' into features/activity-log-record-storage

jdevillard committed:
Merge remote-tracking branch 'origin/main' into features/activity-log-record-storage

sfmskywalker committed:
Merge branch 'main' into features/activity-log-record-storage

sfmskywalker commented on src/modules/Elsa.Studio.Workflows/Components/WorkflowDefinitionEditor/Components/ActivityProperties/Tabs/PersistenceTab.razor:
```diff
@@ -0,0 +1,73 @@
+
+@{
+    
+    RenderFragment GetSelectComponent2(String propertyName, IDictionary<string, LogPersistenceMode> properties) =>
```
Perhaps we can rename this to something else than `GetSelectComponent2`?

jdevillard replied:
Good catch, I've renamed it to `GetSelectComponent`

jdevillard committed:
Merge remote-tracking branch 'origin/features/activity-log-record-storage' into features/activity-log-record-storage

jdevillard committed:
rename GetSelectComponent2 to GetSelectComponent

sfmskywalker committed:
Consolidate project files and update Elsa.Api.Client package

This commit consolidates various project properties files into a single Directory.Build.props file, simplifying the project file structure and reducing duplication. The version of the Elsa.Api.Client package reference is updated and the package is now being directly referenced instead of via a project reference. JetBrains.Annotations usage has been removed as well.

sfmskywalker committed:
Update package version in GitHub workflows

The version of the package in the GitHub workflows has been updated from 3.1.0 to 3.2.0. This was done to ensure consistency with the project's latest release and standards.

sfmskywalker committed:
Merge branch 'main' into features/activity-log-record-storage

sfmskywalker committed:
Refactor code for readability in Settings.razor

The code in Settings.razor has been cleaned up with extraneous spaces removed and alignment of attributes fixed. This modification enhances code readability and maintainability in the workflow properties' settings section.

sfmskywalker committed:
Refactor code formatting in Settings.razor.cs

Adjusted the code formatting to improve readability in Settings.razor.cs. The modifications mainly include adding extra line breaks and removing unnecessary white space. No functionality was changed in this commit.
